server/jetstream.go

```diff
@@ -1556,12 +1556,16 @@ func (a *Account) filteredStreams(filter string) []*stream {
 	var msets []*stream
 	for _, mset := range jsa.streams {
 		if filter != _EMPTY_ {
-			for _, subj := range mset.cfg.Subjects {
-				if SubjectsCollide(filter, subj) {
-					msets = append(msets, mset)
-					break
+			func() {
```

Comment:
Since we're doing a `break` rather than an escaping `return`, we don't really need the anonymous function closure here for `defer`. We can just put the `RUnlock` after the `for` loop instead, which flattens it out a bit.

Reply:
Oop, yes!  I got switched up and read the `break` as in a different context than needs locking, but you're totally correct.

I've squashed the commit, since it doesn't seem that helpful to have one version and then the other.